src/main/java/de/retest/recheck/ui/descriptors/SutState.java

```diff
@@ -78,6 +98,8 @@ public SutState applyChanges( final ActionChangeSet actionChangeSet ) {
 				descriptors.add( (RootElement) element );
 			}
 		}
-		return new SutState( descriptors );
+		final Map<String, String> newMetadata = new HashMap<>( metadata );
+		newMetadata.putAll( actionChangeSet.getMetadata() );
+		return new SutState( descriptors, newMetadata );
```

Comment:
It depends on how the implementation of `ActionChangeSet` handles the metadata. I would argue, that it contains the already merged (i.e. new) metadata (similar to the screenshot, we do no logic here). How is it supposed to delete metadata?

That means, following would suffice:
```
return new SutState( descriptor, actionChangeSet.getMetadata() )
```